lms/djangoapps/courseware/tests/test_date_summary.py

```diff
@@ -214,6 +220,7 @@ def test_enabled_block_types_with_assignments(self):  # pylint: disable=too-many
                 start=now - timedelta(days=14),
                 due=now + timedelta(days=10),
                 graded=True,
+                format='Speech',
```

Comment:
Somewhere in here, you should at least test for the case of `format` being undefined and receiving the default value of `None` from above.

Reply:
I think None only exists when the assignment is ungraded. I think we talked about the default being 'Homework', I'm not sure if that's true but I can do some digging.

Reply:
Okay, in that case, you may want to also update https://github.com/edx/edx-platform/pull/24130/files#diff-fddcdf03b768a768d398d80b8798e76dR545. But yeah, figure out the default case and make sure we still have a test for a graded item without format specified just to make sure everything performs correctly there as well.

Reply:
Oh good catch! I missed that the courses that aren't graded are skipped in this loop.

I just created a new self-paced course to see what the default graded format value is, and we provide four to choose from. The options are 'Homework', 'Lab', 'Midterm Exam', and 'Exam'. If you don't select one, your only other option is 'Not graded' which sets format to `None`. In the line that you linked, do you think changing `None` to `'Homework'` is the best bet? I'm unsure.

Re: testing for `None`, from what I can see either an item is not graded and format is set to `None`, or the item is graded and users are forced to select an assignment type that will set format. I don't believe there's a case where we allow a graded item to have a format type of `None`, unless there's a use case that I'm missing!

Reply:
Unfortunately, I'm not sure we can guarantee the same things due to Courses that use Open Learning XML (OLX) to create/edit courses. They can do basically anything they want there and aren't subject to the limitations of the Studio UI. I think leaving it as none is a fine approach to be honest if we encounter the case of a graded subsection with an unknown format. I just want to make sure we have a test for that case. (And luckily your UI already handles this by not adding in the format unless it is a truthy value)

And quick little correction to your first sentence, we are only dealing with a single Course here with multiple subsections. The `get_course_assignments` (or whatever it is actually called) is only concerned with graded subsections which is why that ungraded one is skipped. But just want to call out that it is subsections we are concerned with, not courses.

Reply:
Ooh, that makes sense! Thanks for that explanation. I've never heard of OLX.

And good catch, you're right, I meant subsections.